src/test/java/seedu/address/storage/JsonAdaptedRecordTest.java

```diff
@@ -0,0 +1,108 @@
+package seedu.address.storage;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static seedu.address.testutil.TypicalPatients.ALICE;
+import static seedu.address.testutil.TypicalPatients.BENSON;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.model.patient.Patient;
+import seedu.address.model.patient.Record;
+
+public class JsonAdaptedRecordTest {
+    private final Patient expectedPatient = BENSON;
+    @Test
+    public void toModelType_validPatientDetails_returnsPatient() throws Exception {
+        String expectedInitialObservations = "Initial observations";
+        String expectedDiagnosis = "Diagnosis details";
+        String expectedTreatmentPlan = "Treatment plan";
+
+        // Create a JsonAdaptedRecord object with the above details
+        JsonAdaptedRecord jsonAdaptedRecord = new JsonAdaptedRecord(
+                expectedPatient, expectedInitialObservations, expectedDiagnosis, expectedTreatmentPlan);
+
+        // Execution
+        Record modelRecord = jsonAdaptedRecord.toModelType();
+
+        // Verification
+        assertEquals(expectedPatient, modelRecord.getPatient());
+        assertEquals(expectedInitialObservations, modelRecord.getInitialObservations());
+        assertEquals(expectedDiagnosis, modelRecord.getDiagnosis());
+        assertEquals(expectedTreatmentPlan, modelRecord.getTreatmentPlan());
+    }
+
+    @Test
+    public void toModelType_emptyFields_createsIncompleteRecord() throws Exception {
+        // Test with empty strings
+        JsonAdaptedRecord jsonAdaptedRecord = new JsonAdaptedRecord(
+                expectedPatient, "", "", "");
+
+        // Execution
+        Record modelRecord = jsonAdaptedRecord.toModelType();
+
+        // Verification
+        assertEquals(expectedPatient, modelRecord.getPatient());
+        assertEquals("", modelRecord.getInitialObservations());
+        assertEquals("", modelRecord.getDiagnosis());
+        assertEquals("", modelRecord.getTreatmentPlan());
+    }
+
```

Comment:
I like how you used the comments to guide developers on the different logics of the method. Good job!